File: be-smart-job/src/main/java/be_smart_job/service/identity/interfaces/UserService.java
package be_smart_job.service.identity.interfaces;

import be_smart_job.dto.req.identity.ChangePasswordRequest;
import be_smart_job.dto.req.identity.UserUpdateRequest;
import be_smart_job.dto.res.identity.UserResponse;
import be_smart_job.enums.UserStatus;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface UserService {
    List<UserResponse> getAllUsers();
    UserResponse getUserById(String id);
    UserResponse getMyProfile();
    UserResponse updateProfile(UserUpdateRequest request);
    UserResponse uploadAvatar(MultipartFile file);
    void changePassword(ChangePasswordRequest request);
    UserResponse updateStatus(String id, UserStatus status);
    void deleteUser(String id);
}

File: fe-smart-job/src/services/userService.js
import axiosClient from '../config/axiosClient';

export const userService = {
  getAllUsers: () => axiosClient.get('/users'),
  
  getUserById: (id) => axiosClient.get(`/users/${id}`),
  
  getMyProfile: () => axiosClient.get('/users/my-profile'),
  
  updateProfile: (data) => axiosClient.put('/users/profile', data),
  
  uploadAvatar: (file) => {
    const formData = new FormData();
    formData.append('file', file);
    return axiosClient.post('/users/avatar', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
  },
  
  changePassword: (data) => axiosClient.put('/users/change-password', data),
  
  updateStatus: (id, status) => axiosClient.patch(`/users/${id}/status`, null, { params: { status } }),
  
  deleteUser: (id) => axiosClient.delete(`/users/${id}`),
};